### PR TITLE
Feature/dest snap delete

### PR DESCRIPTION
**Pull request by snehring**
In progress. Creating this for review.

### CODE REVIEW

**Comment by njbooher** [comment]
There are a lot of commits here. You should squash them. 🍆🥝🥝🥝🥝🥝🥝🥝🍈🥥🥦🥔🥨🍠🍠🥑🥑🥑🥑🥑🥑🍔🍔🍔🍔

**Comment by njbooher** [comment]
Needs more emoji. ⚽️🤡

File: zfsbackup.py [hunk 1]
```diff
@@ -80,6 +80,7 @@ def main():
             lf_path = conf.get('lock_file')
         # TODO future: user selectable logging levels
         logging.getLogger().setLevel(logging.INFO)
+        retain_snaps = conf.get('retain_snaps')
```
Comment (baberlevi) [inline zfsbackup.py:83]: might want to conditionally set this like the other global confs in case it's not defined

File: zfsbackup.py [hunk 2]
```diff
@@ -532,6 +535,107 @@ def has_backuplast(dataset, inc_name):
...
+            for snap in snaps:
+                try:
+                    delete_snapshot(snap)
+                except ZFSBackupError:
+                    errors += 1
+            if errors > 0:
+                logging.warn("Encountered errors while deleting old snapshots" 
+                             + "from destination: "+dataset+" via "
+                             + transport)
+        elif transport.lower().split(':')[0] == 'ssh':
```
Comment (baberlevi) [inline zfsbackup.py:570]: the ssh parsing here seems to also be repeated in verify_backup and send_snapshot - maybe breakout to a function?

File: zfsbackup.py (hunk 1 above)
Comment (snehring) [reply zfsbackup.py:83]: It'll be None if it's not set, which the clean_dest_snaps function uses in conjunction with the lack of destination retain_snaps to decide not to delete any snaps.

File: zfsbackup.py (hunk 2 above)
Comment (snehring) [reply zfsbackup.py:570]: This should be addressed now. There are other areas where code reuse is a thing, but I'd like to save that for a separate cleanup PR.

**Review by isu-rassilon — APPROVED** [review]
Looks good.

**Pull request merged**